[1603] Introduce Agent endpoints and forms with their logic

Struchid opened the pull request:
Introducing Agent endpoints to the platform.

- OneToOne relation with the `ApiKey` model.
   - Agents must have their very own Api Key which would let them access the system, thus the additional relationship on the ApiKey itself, as it is done with Representative model.
- Added endpoints + forms for Agents: Create, Edit, List, Detail & Delete with logic to perform those actions.
- Introducing the additional breadcrumbs, which would show the current page, but not a link, as it was before (in some forms).

Struchid committed:
[1605] Introduce Agent endpoints and forms with their logic

Struchid commented on vitrina/orgs/templates/vitrina/orgs/agents/agent_detail.html:
```diff
@@ -0,0 +1,192 @@
+    </a>
+</p>
+
+<div style="margin-bottom: 2rem; max-width: 800px;">
```
Styles to classes in .css files

Struchid commented on vitrina/orgs/templates/vitrina/orgs/agents/agent_detail.html:
The breadcrumbs should also have some additional space before the ending breadcrumb (which does not link anywhere):

<img width="530" alt="image" src="https://github.com/user-attachments/assets/ea225a21-6dd7-456d-9cee-a39e84683379" />

Struchid replied:
Done:
<img width="654" alt="image" src="https://github.com/user-attachments/assets/a9592da7-e08c-487c-9fc6-84cc55b0dbc6" />

Struchid commented on vitrina/orgs/urls.py:
```diff
@@ -294,10 +299,29 @@
-    # @GetMapping("/report")
-    # @PostMapping("/report")
-    # @GetMapping("/dictionary")
+    path(
```
Tests for endpoints

Struchid replied:
[4ed49f2](https://github.com/atviriduomenys/katalogas/pull/1629/commits/4ed49f210a1d15f7a4a83c0ef07511ce0e7343c4)

Struchid reviewed (commented):
I am leaving additional comments for myself for anything I see that needs doing, but I also want this to be prepared as soon as possible, so it could be reviewed by more people.

martynasracct commented on vitrina/locale/en/LC_MESSAGES/django.po:
```diff
@@ -1786,6 +1786,42 @@ msgstr "Enter the organization's name or the full company code."
 msgid "Naujos organizacijos koordinatorius"
 msgstr "Coordinator for a new organization"
 
+msgid "Nurodomas Agento pavadinimas, unikalus per organizaciją"
```
```suggestion
msgid "Nurodomas Agento pavadinimas, unikalus organizacijai"
```

martynasracct commented on vitrina/locale/en/LC_MESSAGES/django.po:
```diff
@@ -1995,6 +2025,115 @@ msgstr "No, go back"
+
+msgid ""
+"Konfigūracijos failas, kuriame nurodomi prisijungimo parametrai, kurie "
+"nebuvo nurodyti struktūros apraše."
```
```suggestion
msgid ""
"Konfigūracijos failas, kuriame nurodomi prisijungimo parametrai, "
"nenurodyti struktūros apraše."
```

martynasracct commented on vitrina/orgs/forms.py:
```diff
@@ -1310,3 +1311,56 @@ def clean_contact(self):
+
+
+class AgentForm(ModelForm):
+    title = CharField(
```
Question: do we need to override these fields in `ModelForm`, are fields automatically generated from models not enough?

martynasracct commented on vitrina/orgs/views.py:
```diff
@@ -3198,6 +3203,223 @@ class RepresentativeRequestExistsView(TemplateView):
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.object = get_object_or_404(Agent, pk=kwargs.get("pk"), organization=self.organization, is_archived=False)
```
```suggestion
        self.object = get_object_or_404(Agent.objects.select_related("service"), pk=kwargs.get("pk"), organization=self.organization, is_archived=False)
```

martynasracct reviewed (approved):
LGTM. Added some small comments

wisecrow reviewed (approved):
LGTM

Struchid commented:
> LGTM. Added some small comments

Good point with the Form field overrides, removed them, and adjusted the other comments as well. Appended the commit, tho, to not have a lot of unnecessary commits.

Struchid committed:
[1605] Tests

Struchid committed:
[1605] Disable API Key on AGent deletion

Struchid committed:
[1603] Add changes after Sprint Review
